src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+                PreparedStatement statement = connection.prepareStatement(findByIdQuery)) {
+            statement.setLong(KEY_INDEX,id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String findAllQuery = "SELECT * FROM book";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(findAllQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(parseBookFromResultSet(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to select all books form DB", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateQuery = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateQuery)) {
+            statement.setString(TITLE_INDEX, book.getTitle());
+            statement.setBigDecimal(PRICE_INDEX, book.getPrice());
+            statement.setLong(KEY_POSITION_INDEX, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows == EXCEPTION_AFFECTED_ROWS) {
+                throw new RuntimeException("Updating book failed, no rows affected.");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error updating book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String deleteQuery = "DELETE FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(KEY_INDEX, id);
+            int updatedRows = statement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error deleting book by id: " + id, e);
+        }
+    }
+
+    private Book parseBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Book newBook = new Book();
+        newBook.setId(resultSet.getObject(ID_COLUMN, Long.class));
+        newBook.setTitle(resultSet.getObject(TITLE_COLUMN, String.class));
+        newBook.setPrice(resultSet.getObject(PRICE_COLUMN, BigDecimal.class));

```

Comment:
in the case of `String` and `BigDecimal`, it is actually fine to use `getString()` and `.getBigDecimal()` because it is already an object, when with case of `getLong()` it returns a primitive(which can not be `null`) and thus we have to use `getObject()` instead.
It's fine to leave it as it is, just wanted you to know 😉